### PR TITLE
docs: workflow error handling documentation

### PR DESCRIPTION
**Pull request by aludwiko**

### CODE REVIEW

File: docs/src/modules/java/pages/workflow-entities.adoc [hunk 1]
```diff
@@ -96,16 +96,77 @@ include::example$java-spring-transfer-workflow/src/main/java/com/example/transfe
...
+=== Recover strategy
+
+It's time to define what should happen in case of timeout or any other unhandled error.
+
+[source,java,indent=0]
+.src/main/java/com/example/transfer/TransferWorkflow.java
+----
+include::example$java-spring-transfer-workflow-compensation/src/main/java/com/example/transfer/TransferWorkflow.java[tag=recover-strategy]
+----
+<1> Sets a failover transition in case of a workflow timeout
```
Comment (efgpinto) [inline docs/src/modules/java/pages/workflow-entities.adoc:145]:
```suggestion
<1> Sets a failover transition in case of a workflow timeout.
```

**Review by efgpinto — APPROVED** [review]
LGTM

File: docs/src/modules/java/pages/workflow-entities.adoc [hunk 2]
```diff
@@ -96,16 +96,77 @@ include::example$java-spring-transfer-workflow/src/main/java/com/example/transfe
...
+<2> Sets a default timeout for all workflow steps.
+
+A default step timeout can be overridden in step builder.
+
+[source,java,indent=0]
+.src/main/java/com/example/transfer/TransferWorkflow.java
+----
+include::example$java-spring-transfer-workflow-compensation/src/main/java/com/example/transfer/TransferWorkflow.java[tag=step-timeout]
+----
+<1> Overrides a step timeout for a specific step.
```
Comment (octonato) [inline docs/src/modules/java/pages/workflow-entities.adoc:134]:
```suggestion
<1> Overrides the step timeout for a specific step.
```

File: docs/src/modules/java/pages/workflow-entities.adoc [hunk 3]
```diff
@@ -96,16 +96,77 @@ include::example$java-spring-transfer-workflow/src/main/java/com/example/transfe
...
+<1> Explicit deposit call result type.
+<2> Finishes workflow as completed, in the case of a successful deposit.
+<3> Launches compensation step to handle deposit failure. A withdraw step must be reversed.
+<4> Compensation step is just any other step, with the same set of functionalities.
+<5> Correct compensation can finish the workflow.
+<6> Any other result might be handled by a default recovery strategy.
+
+Compensating a workflow step (steps) might involve multiple logical steps and thus is part of the overall business logic that must be defined within the workflow itself. For simplicity, in the example above, the compensation is applied only to `withdraw` step. Whereas `deposit` step itself might also require a compensation. In case of a step timeout we can't be certain about step successful or error outcome.
+
+//TODO fix url
```
Comment (octonato) [inline docs/src/modules/java/pages/workflow-entities.adoc:170]: which url?

File: samples/java-spring-transfer-workflow-compensation/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# If you're looking to use eventing with Google PubSub, to get an emulator running:
+# - uncomment property "-Dkalix.proxy.eventing.support=google-pubsub-emulator"
+# - uncomment the env var PUBSUB_EMULATOR_HOST and the section below for gcloud-pubsub-emulator service
+version: "3"
+services:
+  kalix-proxy:
+    image: gcr.io/kalix-public/kalix-proxy:1.1.5
```
Comment (octonato) [inline samples/java-spring-transfer-workflow-compensation/docker-compose.yml:7]:
```suggestion
    image: gcr.io/kalix-public/kalix-proxy:1.1.6
```

File: samples/java-spring-transfer-workflow-compensation/pom.xml
Comment (octonato) [inline samples/java-spring-transfer-workflow-compensation/pom.xml:1]: This pom.xml is an old one. The new version has `spring-boot-parent` and the shading plugin is under `pluginManagement`. Probably easier to copy from another sample and change the name and artifactId

**Review by octonato — APPROVED** [review]
LGTM

With a some minor suggestions and change request. 

Ok for merge after that.

File: docs/src/modules/java/pages/workflow-entities.adoc (hunk 3 above)
Comment (aludwiko) [reply docs/src/modules/java/pages/workflow-entities.adoc:170]: the one below, I will fix it after I merge this change since this repo does not exist yet.

**Pull request merged**